agent/consul/rate/handler_test.go

```diff
@@ -224,10 +257,67 @@ func TestHandler(t *testing.T) {
 			} else {
 				require.Zero(t, output.Len(), "expected no logs to be emitted")
 			}
+
+			if tc.expectMetric {
+				assertCounter(t, sink, tc.expectMetricName, tc.expectMetricCount)
+			}
 		})
 	}
 }
 
+func assertCounter(t *testing.T, sink *metrics.InmemSink, name string, value float64) {
+	t.Helper()
+
+	data := sink.Data()
+
+	var got float64
+	for _, intv := range data {
+		intv.RLock()
+		// Note that InMemSink uses SampledValue and treats the _Sum_ not the Count
+		// as the entire value.
+		if sample, ok := intv.Counters[name]; ok {
+			got += sample.Sum
+		}
+		intv.RUnlock()
+	}
+
+	if !assert.Equal(t, value, got) {
+		// no nice way to dump this - this is copied from private method in
+		// InMemSink used for dumping to stdout on SIGUSR1.
+		buf := bytes.NewBuffer(nil)
+		for _, intv := range data {
+			intv.RLock()
+			for name, val := range intv.Gauges {
+				fmt.Fprintf(buf, "[%v][G] '%s': %0.3f\n", intv.Interval, name, val.Value)
+			}
+			for name, vals := range intv.Points {
+				for _, val := range vals {
+					fmt.Fprintf(buf, "[%v][P] '%s': %0.3f\n", intv.Interval, name, val)
+				}
+			}
+			for name, agg := range intv.Counters {
+				fmt.Fprintf(buf, "[%v][C] '%s': %s\n", intv.Interval, name, agg.AggregateSample)
+			}
+			for name, agg := range intv.Samples {
+				fmt.Fprintf(buf, "[%v][S] '%s': %s\n", intv.Interval, name, agg.AggregateSample)
+			}
+			intv.RUnlock()
+		}
+		t.Log(buf.String())
+	}
+}
+
+func testSetupMetrics(t *testing.T) *metrics.InmemSink {
+	// Record for ages (5 mins) so we can be confident that our assertions won't
+	// fail on silly long test runs due to dropped data.
+	s := metrics.NewInmemSink(10*time.Second, 300*time.Second)
+	cfg := metrics.DefaultConfig("")
+	cfg.EnableHostname = false
+	cfg.EnableRuntimeMetrics = false
+	metrics.NewGlobal(cfg, s)
+	return s
+}
```

Comment:
I copy/pasted this and `assertCounter(...)` from `listener_test.go`. Is there a good place to move and share common test utility functions like these?